### DISCUSSION TITLE
How the best way to use TFlite to check how correct an expected word was spoken?

### DISCUSSION THREAD

**Discussion by echogit** [tensorflow/flutter-tflite#212] [General]
Îs there a suggested way to use tflite to give a score from 0-100 to how well a word has been spoken?

I thought in training a model using the correct word being said correctly, probably even using TTS and after getting words being cut or badly spoken and give them lower scores. Just as a high level solution. Is there something ready for that?
My end goal is to be able to ask for a children say a word, and depending how well it was said, give him/her a positive feedback.

**Comment by PaulTR** [comment]
That's a fun idea! So I have no idea how well it'd work, but there's a
concept technique called embedding using cosine similarities that basically
tries to determine how closely two different pieces of data are matched up.
Between audio classification confidence scores and that you might be on the
right track. Unfortunately I'm away from my laptop right now (and honestly
swamped with IO stuff even if I were at it :p) or I'd see if I could get
you more information, but hopefully that's a start.
